vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/per_day_frequency_cap_unittest.cc

```diff
@@ -0,0 +1,178 @@
+    // destructor)
+  }
+
+  std::unique_ptr<MockAdsClient> mock_ads_client_;
+  std::unique_ptr<AdsImpl> ads_;
+
+  std::unique_ptr<ClientMock> client_mock_;
+  std::unique_ptr<FrequencyCapping> frequency_capping_;
+  std::unique_ptr<PerDayFrequencyCap> exclusion_rule_;
+  AdInfo ad_info_;
+};
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdAllowedWhenNoAds) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdAllowedBelowDailyCap) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, 0, 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdAllowedWithAdOutsideDayWindow) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+  // Now
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, 0, 1);
+  // 24hrs + 1s ago
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, -kSecondsPerDay, 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_FALSE(is_ad_excluded);
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdExcludedAboveDailyCap1) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, 0, 2);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_TRUE(is_ad_excluded);
+  EXPECT_EQ(exclusion_rule_->GetLastMessage(), "creativeSetId 654f10df-fbc4-4a92-8d43-2edf73734a60 has exceeded the frequency capping for perDay");  // NOLINT
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdExcludedAboveDailyCap2) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 2;
+
+  // Now
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, 0, 1);
+  // 23hrs 59m 59s ago
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, -(kSecondsPerDay - 1), 1);
+
+  // Act
+  bool is_ad_excluded = exclusion_rule_->ShouldExclude(ad_info_);
+
+  // Assert
+  EXPECT_TRUE(is_ad_excluded);
+  EXPECT_EQ(exclusion_rule_->GetLastMessage(), "creativeSetId 654f10df-fbc4-4a92-8d43-2edf73734a60 has exceeded the frequency capping for perDay");  // NOLINT
+}
+
+TEST_F(BraveAdsPerDayFrequencyCapTest, AdExcludedForIssue4207) {
+  // Arrange
+  ad_info_.creative_set_id = test_creative_set_id;
+  ad_info_.per_day = 20;
+
+  // 5 ads per hour, up to a total of 20
+  client_mock_->GenerateCreativeSetHistory(
+      test_creative_set_id, -(base::Time::kSecondsPerHour / 5), 20);
```

Comment:
What if the default of 5 ads per hour or 20 ads per day is changed, can you please mock for each set of unit tests:

GetAdsPerHour` to return `5` as follows:
```
    ON_CALL(*mock_ads_client_, GetAdsPerHour())
        .WillByDefault(Return(5));
```
`GetAdsPerDay` to `20` as follows:
```
    ON_CALL(*mock_ads_client_, GetAdsPerDay())
        .WillByDefault(Return(20));
```

Either add to setup so all tests work off the same criteria or to individual tests if you test different values

Note: you will also need `using ::testing::Return;`

Reply:
Happy to oblige, but this is a specific test for the specific criteria mentioned in the issue 4207, and neither of those calls are made during the test so mocking won't help? Or I'm not sure what you mean...

Reply:
Yeap missed that :-)